database/factories/ModelFactory.php

```diff
@@ -104,9 +105,52 @@
 $factory->defineAs(App\Voucher::class, 'requested', function ($faker) use ($factory) {
     $voucher = $factory->raw(App\Voucher::class);
 
-    // Todo create the voucher_states on the road to requested.
-
     return array_merge($voucher, [
         'currentstate' => 'requested',
     ]);
 });
+
+/**
+ * Voucher with currentstate allocated.
+ */
+$factory->defineAs(App\Voucher::class, 'allocated', function ($faker) use ($factory) {
```

Comment:
To make writing tests easier - I can't instatiate the voucher inside the factory that creates it - so I have to add the states manually or do it in a Seeder or Test as I did for the Voucher Seeder. Did you also see that there seems to be a bug with the StateMachine or the way we are using it. VoucherStates don't seem to be recording from and source values.